store/store.go

```diff
@@ -670,7 +673,7 @@ func (s *SnapUbuntuStoreRepository) Download(remoteSnap *snap.Info, pbar progres
 	if err != nil {
 		return "", err
 	}
-	s.setUbuntuStoreHeaders(req, "", auther)
+	s.setUbuntuStoreHeaders(req, "", remoteSnap.NeedsDevMode(), auther)
```

Comment:
The naming looks suspect here. Is that the local snap or the remote snap? It doesn't make sense to send devmode=true just because the remote snap claims to be in devmode, right? It's the local state that we're trying to report to the store, right?